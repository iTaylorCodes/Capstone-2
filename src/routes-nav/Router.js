import { Route, Routes, Navigate } from "react-router-dom";
import Home from "../homepage/Home";
import SignupForm from "../auth/SignupForm";
import LoginForm from "../auth/LoginForm";
import Profile from "../profile/Profile";
import Neighborhood from "../neighborhoods/Neighborhood";
import FavoritesList from "../favorites/FavoritesList";

const Router = () => {
  return (
    <Routes>
      <Route path="*" element={<Navigate replace to="/" />} />
      <Route path="/" element={<Home />} />
      <Route path="/signup" element={<SignupForm />} />
      <Route path="/login" element={<LoginForm />} />
      <Route path="/profile" element={<Profile />} />
      <Route path="/neighborhood/:neighborhood" element={<Neighborhood />} />
      <Route path="/favorites" element={<FavoritesList />} />
    </Routes>
  );
};

export default Router;
