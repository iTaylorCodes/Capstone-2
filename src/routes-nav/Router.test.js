import { render } from "@testing-library/react";
import Router from "./Router";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <Router />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <Router />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
