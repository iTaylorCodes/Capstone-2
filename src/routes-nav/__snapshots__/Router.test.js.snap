// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center home-hero"
  >
    <div
      class="fw-bold"
      id="alert-container"
    />
    <div
      class="HomeContent"
    >
      <div
        class="Home-heading m-5"
      >
        <h1
          class="text-primary mb-3"
        >
          <img
            alt="Logo"
            class="me-2 Home-logo"
            src="Neighborhoodz-Logo.png"
          />
          Neighborhoodz
        </h1>
        <h3>
          Find your new Neighborhood now!
        </h3>
      </div>
      <div
        class="SearchBar d-flex justify-content-center"
      >
        <div
          class="mb-3"
        >
          <form
            class="input-group"
          >
            <div
              class="form-floating"
            >
              <input
                class="form-control ps-5 pe-5"
                id="city-search"
                name="searchTerm"
                placeholder="Search City, State:"
                value=""
              />
              <label
                for="city-search"
              >
                Search City, State:
              </label>
            </div>
            <button
              class="btn btn-primary"
              data-testid="search-button"
              type="submit"
            >
              <i
                class="fa fa-search"
              />
            </button>
          </form>
        </div>
      </div>
      <div
        class="SearchesCarousel container d-flex justify-content-center"
      >
        <div
          class="col-12 col-md-8 col-lg-6 col-xl-5 col-xxl-4 bg-light rounded text-center p-1"
        >
          <h3>
            Recent Searches
          </h3>
          <p>
            Your most recent searches will be shown here.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
